### PR TITLE
Use a shared HTTP client implementation

### PR DESCRIPTION
**Pull request by nlordell**
Use a shared HTTP client implementation. This is the first step for #581 so we can share metrics across HTTP services.

### Test Plan

CI

### CODE REVIEW

**Review by e00E — APPROVED** [review]

File: driver/src/http/mod.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use isahc::http::{HttpTryFrom, Uri};
+use isahc::prelude::Request;
+use isahc::{HttpClientBuilder, ResponseExt};
+use serde::de::DeserializeOwned;
+use std::time::Duration;
+
+#[cfg(test)]
+lazy_static::lazy_static! {
+    /// An HTTP factory used for testing.
+    pub static ref TEST_FACTORY: HttpFactory = HttpFactory::new(Duration::from_secs(10));
```
Comment (fleupold) [inline driver/src/http/mod.rs]: I'd expect a test factory to return mocked versions of HTTPClient (to set expectations without doing network calls). I guess this is a real factory used in the #[ignore] tests. Not sure if the name could capture this somehow. Would it maybe make sense to implement Default on HttpFactory?

**Review by fleupold — APPROVED** [review]
Any particular reason we want to use multiple clients instead of a single one?

File: driver/src/http/mod.rs (hunk 1 above)
Comment (nlordell) [reply driver/src/http/mod.rs]: Maybe we can rename this to `GLOBAL_FACTORY`?. This is really intended for creating real actual HTTP clients for the ignored tests.

**Comment by nlordell** [comment]
> Any particular reason we want to use multiple clients instead of a single one?

Its so they can have separate configurations.

File: driver/src/http/mod.rs (hunk 1 above)
Comment (nlordell) [reply driver/src/http/mod.rs]: Just implemented `Default`, it makes the most sense.

**Comment by nlordell** [comment]
@e00E do you mind taking a quick look at the changes I made to the DexAG code and let me know if you think they are OK? This changes it to use the new HTTP factory.

File: driver/src/price_estimation/dexag/api.rs [hunk 2]
```diff
@@ -80,16 +75,14 @@ impl DexagApi for DexagHttpApi {
...
-            })
-            .boxed()
+        async move {
+            Ok(self
+                .client
+                .get_json_async(url.as_str())
+                .await
+                .context("failed to parse price json from dexag")?)
+        }
+        .boxed()
```
Comment (e00E) [inline driver/src/price_estimation/dexag/api.rs]: There is a problem here.

This function here returns f64 so `get_json_async` tries to deserialize into f64 but it should deserialize into `Price`.

```
async move {
            Ok(self
                .client
                .get_json_async::<_, Price>(url.as_str())
                .await
                .context("failed to parse price json from dexag")?
                .price)
        }
        .boxed()
```

Comment (e00E) [reply driver/src/price_estimation/dexag/api.rs]: That is the only problem I see.

Comment (nlordell) [reply driver/src/price_estimation/dexag/api.rs]: Fixed :)

**Pull request merged**